timbuctoo-instancev4/src/main/java/nl/knaw/huygens/timbuctoo/core/AfterSuccessTaskExecutor.java

```diff
@@ -3,15 +3,11 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.LinkedList;
+import java.util.ArrayDeque;
 
 public class AfterSuccessTaskExecutor {
   public static final Logger LOG = LoggerFactory.getLogger(AfterSuccessTaskExecutor.class);
-  private final LinkedList<Task> tasks;
-
-  public AfterSuccessTaskExecutor() {
-    tasks = new LinkedList<>();
-  }
+  private final ArrayDeque<Task> tasks = new ArrayDeque<>();
```

Comment:
1. Deque implementations are the idiomatic way to represent stacks and queues in Java.
2. Arrays perform better than linked lists (locality of reference, less GC pressure).

(1.) is the main reason here.